pkcs10/src/attribute.rs

```diff
@@ -0,0 +1,40 @@
+use der::asn1::{Any, ObjectIdentifier, SetOfVec};
+use der::{Decodable, OrdIsValueOrd, Sequence};
+
+/// PKCS#10 `Attribute` as defined in [RFC 2986 Section 4].
+///
+/// ```text
+/// Attribute { ATTRIBUTE:IOSet } ::= SEQUENCE {
+///     type   ATTRIBUTE.&id({IOSet}),
+///     values SET SIZE(1..MAX) OF ATTRIBUTE.&Type({IOSet}{@type})
+/// }
+/// ```
+///
+/// [RFC 2986 Section 4]: https://datatracker.ietf.org/doc/html/rfc2986#section-4
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Sequence)]
```

Comment:
```suggestion
#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Sequence, ValueOrd)]
```

Reply:
Attempting to do this produces the following error:

```
error[E0599]: the method `der_cmp` exists for struct `SetOfVec<der::Any<'a>>`, but its trait bounds were not satisfied
   --> pkcs10/src/attribute.rs:14:66
    |
14  |   #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Sequence, ValueOrd)]
    |                                                                    ^^^^^^^^ method cannot be called on `SetOfVec<der::Any<'a>>` due to unsatisfied trait bounds
    |
   ::: /Users/npmccallum/Projects/formats/der/src/asn1/set_of.rs:151:1
    |
151 | / pub struct SetOfVec<T>
152 | | where
153 | |     T: Clone + DerOrd,
154 | | {
155 | |     inner: Vec<T>,
156 | | }
    | | -
    | | |
    | |_doesn't satisfy `SetOfVec<der::Any<'_>>: DerOrd`
    |   doesn't satisfy `SetOfVec<der::Any<'_>>: ValueOrd`
    |
    = note: the following trait bounds were not satisfied:
            `SetOfVec<der::Any<'_>>: ValueOrd`
            which is required by `SetOfVec<der::Any<'_>>: DerOrd`
    = note: this error originates in the derive macro `ValueOrd` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0599`.
error: could not compile `pkcs10` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Reply:
Aah, the problem here is the `der::asn1::SetOfVec` type itself is missing a `ValueOrd` impl. Once that's added it will work.

Reply:
@tarcieri I just tried that. The derive macro doesn't seem to work. And I'm not sure what this trait actually does. So implementing it manually is a bit confusing.

Reply:
@tarcieri The latest push has a manual `impl` for this. I hope I've done it right. Please take a look.

Reply:
#362 adds the necessary impls to the `der` crate

Reply:
@tarcieri Rebased.